packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -602,103 +629,33 @@ export async function listAirnodes(cloudProviders: readonly CloudProvider['type'
 
-        for (const [stage, stageDirectory] of Object.entries(addressDirectory.children)) {
-          if (deployment) break;
-
-          if (stageDirectory.type !== FileSystemType.Directory) {
-            logger.warn(
-              `Invalid item in bucket '${bucketName}' (${cloudProvider.toUpperCase()}) with key '${
-                stageDirectory.bucketKey
-              }'. Skipping.`
-            );
-            continue;
-          }
-
-          const latestDeployment = Object.keys(stageDirectory.children).sort().reverse()[0];
-          const bucketConfigPath = `${airnodeAddress}/${stage}/${latestDeployment}/config.json`;
-          const config = JSON.parse(
-            await cloudProviderLib[cloudProvider].getFileFromBucket(bucketName, bucketConfigPath)
-          ) as Config;
-          const region = (config.nodeSettings.cloudProvider as CloudProvider).region;
-          const airnodeVersion = config.nodeSettings.nodeVersion;
-          const id = hashDeployment(cloudProvider, region, airnodeAddress, stage, airnodeVersion);
-
-          if (id !== deploymentId) continue;
-
-          const deploymentVersions = Object.keys(stageDirectory.children).map((versionTimestamp) => ({
-            id: hashDeploymentVersion(cloudProvider, region, airnodeAddress, stage, airnodeVersion, versionTimestamp),
-            timestamp: versionTimestamp,
-          }));
-          deployment = {
-            id,
-            cloudProvider,
-            region,
-            airnodeAddress,
-            stage,
-            airnodeVersion,
-            lastUpdate: latestDeployment,
-            versions: deploymentVersions,
-          };
-        }
-      }
-    });
-
-    if (!goCloudDeploymentInfo.success) {
-      // TODO: Again, this spinner mess will be fixed once https://github.com/api3dao/airnode/issues/1473 is done
-      spinner.stop();
-      logger.fail(`Failed to fetch deployment info from ${cloudProvider.toUpperCase()}`);
-      spinner.start(`Fetching info about deployment '${deploymentId}'`);
-      if (logger.inDebugMode()) {
-        spinner.info();
-      }
-    }
+  if (!goCloudDeploymentInfo.success) {
+    spinner.stop();
+    throw new Error(
+      `Failed to fetch deployment info from ${cloudProvider.toUpperCase()}: ${goCloudDeploymentInfo.error}`
+    );
   }
 
-  if (!deployment) {
-    const message = `No deployment with id '${deploymentId}' found`;
-    spinner.fail(message);
-    throw new Error(message);
+  if (goCloudDeploymentInfo.data.length === 0) {
+    spinner.stop();
+    throw new Error(`No deployment with id '${deploymentId}' found`);
   }
 
-  const { id, cloudProvider, region, airnodeAddress, stage, airnodeVersion, lastUpdate, versions } =
-    deployment as Deployment;
+  const deployment = goCloudDeploymentInfo.data[0];
+  const { id, region, airnodeAddress, stage, airnodeVersion, lastUpdate, versions } = deployment as Deployment;
```

Comment:
Why is this not typed automatically? The go utils should provide inference for these things.